cirq/ops/common_channels.py

```diff
@@ -31,109 +31,196 @@
-        probabilities must be valid probabilities and the sum p_x + p_y + p_z
-        must be a valid probability or else this constructor will raise a
-        ValueError.
+        This channel applies one of 4**n disjoint possibilities: nothing (the
+        identity channel) or one of the 4**n pauli gates. The disjoint
+        probabilities of the 4**n gates.
 
         This channel evolves a density matrix via
 
             $$
-            \rho \rightarrow (1 - p_x - p_y - p_z) \rho
-                    + p_x X \rho X + p_y Y \rho Y + p_z Z \rho Z
+            \sum_i p_i Pi \rho Pi
             $$
 
+        Where i varies from 0 to 4**n-1, representing the 4**n Pauli gates,
+        represented by Pi. The input \rho is the density matrix before the
+        depolarization.
+
         Args:
             p_x: The probability that a Pauli X and no other gate occurs.
             p_y: The probability that a Pauli Y and no other gate occurs.
             p_z: The probability that a Pauli Z and no other gate occurs.
+            num_qubits: Number of qubits
+            error_probabilities: Dictionary of string (Pauli operator) to its
+                probability
 
         Raises:
             ValueError: if the args or the sum of args are not probabilities.
         """
-
-        self._p_x = value.validate_probability(p_x, 'p_x')
-        self._p_y = value.validate_probability(p_y, 'p_y')
-        self._p_z = value.validate_probability(p_z, 'p_z')
-        self._p_i = 1 - value.validate_probability(p_x + p_y + p_z,
-                                                   'p_x + p_y + p_z')
+        if error_probabilities is not None:
+            for k in error_probabilities.keys():
+                if len(k) != num_qubits:
+                    raise ValueError('{} does not have {} entries'.format(
+                        k, num_qubits))
+            self._num_qubits = num_qubits
+            self._error_probabilities = error_probabilities
+        else:
+            if p_x is None:
+                p_x = 0.0
+            if p_y is None:
+                p_y = 0.0
+            if p_z is None:
+                p_z = 0.0
+
+            p_x = value.validate_probability(p_x, 'p_x')
+            p_y = value.validate_probability(p_y, 'p_y')
+            p_z = value.validate_probability(p_z, 'p_z')
+            p_i = 1 - value.validate_probability(p_x + p_y + p_z,
+                                                 'p_x + p_y + p_z')
+
+            self._num_qubits = 1
+            self._error_probabilities = {'I': p_i, 'X': p_x, 'Y': p_y, 'Z': p_z}
 
     def _mixture_(self) -> Sequence[Tuple[float, np.ndarray]]:
-        return ((self._p_i, protocols.unitary(identity.I)),
-                (self._p_x, protocols.unitary(pauli_gates.X)),
-                (self._p_y, protocols.unitary(pauli_gates.Y)),
-                (self._p_z, protocols.unitary(pauli_gates.Z)))
+        Pis = []
```

Comment:
nit: We follow a convention that uses lower case letters for variable names. Perhaps just call it `ps` (this looks like plural of `p` which is the mathematical convention for probability).